From 63e3e71246b24134efb4f181948aa266dd279d99 Mon Sep 17 00:00:00 2001
From: Karthik Prasanna N <quic_kartn@quicinc.com>
Date: Mon, 24 Jun 2024 16:00:03 +0530
Subject: [PATCH] Netfilter: Set int_pri value in skb

Setting the int_pri field in the sk buffer for the packets.

Change-Id: Ib2e3b00c9fd2d44467bc87a19b25df435320d911
Signed-off-by: Karthik Prasanna N <quic_kartn@quicinc.com>
---
 include/uapi/linux/netfilter/xt_CLASSIFY.h | 7 +++++++
 net/netfilter/xt_CLASSIFY.c                | 7 ++++++-
 2 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/include/uapi/linux/netfilter/xt_CLASSIFY.h b/include/uapi/linux/netfilter/xt_CLASSIFY.h
index fe059a98c1d3..af1d8cfbf751 100644
--- a/include/uapi/linux/netfilter/xt_CLASSIFY.h
+++ b/include/uapi/linux/netfilter/xt_CLASSIFY.h
@@ -4,8 +4,15 @@
 
 #include <linux/types.h>
 
+enum {
+	XT_SET_PRIORITY = 1 << 0,
+	XT_SET_INT_PRI = 1 << 1
+};
+
 struct xt_classify_target_info {
 	__u32 priority;
+	__u8 int_pri;
+	__u8 set_flags;
 };
 
 #endif /*_XT_CLASSIFY_H */
diff --git a/net/netfilter/xt_CLASSIFY.c b/net/netfilter/xt_CLASSIFY.c
index 0accac98dea7..557b6f4bf0a0 100644
--- a/net/netfilter/xt_CLASSIFY.c
+++ b/net/netfilter/xt_CLASSIFY.c
@@ -30,7 +30,12 @@ classify_tg(struct sk_buff *skb, const struct xt_action_param *par)
 {
 	const struct xt_classify_target_info *clinfo = par->targinfo;
 
-	skb->priority = clinfo->priority;
+	if (XT_SET_PRIORITY & clinfo->set_flags)
+		skb->priority = clinfo->priority;
+
+	if (XT_SET_INT_PRI & clinfo->set_flags)
+		skb->int_pri = clinfo->int_pri;
+
 	return XT_CONTINUE;
 }
 
-- 
2.34.1

